src/datasets/inspect.py

```diff
@@ -154,21 +156,36 @@ def get_dataset_infos(
             Optional path to the directory in which the dynamic modules are saved. It must have been initialized with :obj:`init_dynamic_modules`.
             By default the datasets and metrics are stored inside the `datasets_modules` module.
         data_files (:obj:`Union[Dict, List, str]`, optional): Defining the data_files of the dataset configuration.
-        download_kwargs: optional attributes for DownloadConfig() which will override the attributes in download_config if supplied,
-            for example ``use_auth_token``
+        use_auth_token (``str`` or ``bool``, optional): Optional string or boolean to use as Bearer token for remote files on the Datasets Hub.
+            If True, will get token from `"~/.huggingface"`.
+        download_kwargs: optional attributes for DownloadConfig() which will override the attributes in download_config if supplied
     """
-    dataset_module = dataset_module_factory(
-        path,
+    config_names = get_dataset_config_names(
+        path=path,
         revision=revision,
         download_config=download_config,
         download_mode=download_mode,
+        data_files=data_files,
         force_local_path=force_local_path,
         dynamic_modules_path=dynamic_modules_path,
-        data_files=data_files,
         **download_kwargs,
     )
-    builder_cls = import_main_class(dataset_module.module_path, dataset=True)
-    return builder_cls.get_all_exported_dataset_infos()
+    return {
+        config_name: get_dataset_config_info(
+            path=path,
+            revision=revision,
+            download_config=download_config,
+            download_mode=download_mode,
+            data_files=data_files,
+            config_name=config_name,
+            data_dir=None,
+            cache_dir=None,
+            features=None,
+            use_auth_token=use_auth_token,
+            # **download_kwargs,
```

Comment:
You can remove this line

Reply:
I think we need at least those ones: path/revision/config_name (obviously), download_config/download_mode_use_auth_token (it downloads the dataset script from the hub), data_files/data_dir (splits may depend on these)

So maybe these two could be potentially removed:
- cache_dir: where to download data files and arrow files (not necessary when we just check the splits or the configs)
- features: overwrite the default feature type of the dataset (completely useless in this case)

what do you think ?

Reply:
data_dir: in this function (get_dataset_infos) it does not make sense either, right, since it's related to a config?

Reply:
My point the mismatch between the two function signatures:

```
def get_dataset_infos(
    path: str,
    revision: Optional[Union[str, Version]] = None,
    download_config: Optional[DownloadConfig] = None,
    download_mode: Optional[GenerateMode] = None,
    force_local_path: Optional[str] = None,
    dynamic_modules_path: Optional[str] = None,
    data_files: Optional[Union[Dict, List, str]] = None,
    use_auth_token: Optional[Union[bool, str]] = None,
    **download_kwargs,
):
```

and 

```
def get_dataset_config_info(
    path: str,
    config_name: Optional[str] = None,
    data_dir: Optional[str] = None,
    data_files: Optional[Union[str, Sequence[str], Mapping[str, Union[str, Sequence[str]]]]] = None,
    cache_dir: Optional[str] = None,
    features: Optional[Features] = None,
    download_config: Optional[DownloadConfig] = None,
    download_mode: Optional[GenerateMode] = None,
    revision: Optional[Union[str, Version]] = None,
    use_auth_token: Optional[Union[bool, str]] = None,
    **config_kwargs,
) -> DatasetInfo:
```

The first one calls the second one

Reply:
I'll check what parameters we need and update the PR, will ping you once it's done :)
Thanks for raising this concern

Reply:
thank you!